### PR TITLE
[MIG] mrp_production_add_middle_stuff: migrated from odoomrp-wip 8.0

### PR DESCRIPTION
**Pull request by matt454357**
This PR moves the 8.0 version of the mrp_production_add_middle_stuff module from odoomrp-wip to OCA manufacture.

I have already done the work of migrating this module to 9.0.  So, as soon as this is merged into OCA, I will make a PR for 9.0 (assuming I have understood the process).

### CODE REVIEW

File: mrp_production_add_middle_stuff/wizard/wizard_addition.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                         'production_id': production.id, 'name': product.name,
+                         'addition': True}
+        return addition_vals
+
+    @api.multi
+    def add_product(self):
+        move_obj = self.env['stock.move']
+        if self.product_qty <= 0:
+            raise exceptions.Warning(
+                _('Warning'), _('Please provide a positive quantity to add'))
```
Comment (elicoidal) [inline mrp_production_add_middle_stuff/wizard/wizard_addition.py]: Quantity must be positive

File: mrp_production_add_middle_stuff/wizard/wizard_addition.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class WizProductionProductLine(models.TransientModel):
+    _name = 'wiz.production.product.line'
+
+    product_id = fields.Many2one('product.product', 'Product', required=True)
+    product_qty = fields.Float(
+        'Product Quantity', digits=dp.get_precision('Product Unit of Measure'),
+        required=True)
+    production_id = fields.Many2one(
+        'mrp.production', 'Production Order', select=True,
```
Comment (elicoidal) [inline mrp_production_add_middle_stuff/wizard/wizard_addition.py]: s/Production Order/Manufacturing Order

File: mrp_production_add_middle_stuff/wizard/wizard_addition.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (elicoidal) [inline mrp_production_add_middle_stuff/wizard/wizard_addition.py]: * Please use simple headers in all py files following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: mrp_production_add_middle_stuff/README.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+MRP Production add middle stuff
```
Comment (elicoidal) [inline mrp_production_add_middle_stuff/README.rst:2]: Main problem I have with this module is the naming
mrp_production_add_raw_material?

File: mrp_production_add_middle_stuff/__init__.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (elicoidal) [inline mrp_production_add_middle_stuff/__init__.py]: use simplify headers for the python files

**Comment by nhomar** [comment]
Can you rebase it please!?

**Comment by nhomar** [comment]
No CLA.

**Comment by nhomar** [comment]
Can you improve the readme please?

**Comment by pedrobaeza** [comment]
You can't name it _add_raw_material, because this doesn't reflect what it does. It is called _middle_stuff_ because it allows to add products to consume in the middle of the production.

**Comment by elicoidal** [comment]
I understand the logic of the module but current name is terrible

**Comment by pedrobaeza** [comment]
The only alternative I see is _mrp_production_add_middle_raw_material_

**Comment by anajuaristi** [comment]
Please don't remember it again.
Initially it was add_row_material but it was changed after a lot of
companies were using it on production causing more than one headeache
As today it's installed in %100 of our manufacturing customer the impact
will be even bigger.
For me, name is rigth since it reflect what module does and people using it
knows that name so please don't change it.
Thank you:
Ana

**Comment by anajuaristi** [comment]
Sorry. Not remember.. I wanted to say Don't rename :)

**Comment by matt454357** [comment]
Can we come to an agreement on the name?   If, as Ana says, the name is appreciated by many users, then let's keep it.

It sounds like the bad name helps users avoid confusion.  The bad name is intentional, and has a purpose.

**Comment by matt454357** [comment]
@nhomar, what do you want from the rebase? Do you want to squash the history? I thought we wanted to keep the history.

**Comment by elicoidal** [comment]
I will not block this PR on naming due to historical reasons

**Comment by matt454357** [comment]
Does this PR satisfy everyone?

**Comment by pedrobaeza** [comment]
Please remove the last 2 commits, but :+1:

**Comment by matt454357** [comment]
I removed the last 2 commits.  Can we get this merged?

**Comment by pedrobaeza** [comment]
We need at least another approval for this to get merged.

**Comment by Daniel-CA** [comment]
Please add tests.
Even if the reviewer has been Pedro, this code is mine, just to clarify data on license signature.

**Comment by LoisRForgeFlow** [comment]
Closing as this is stale and it has comments not answered by the author. Please reopen if you think still is relevant.

**Pull request closed without merging**